src/socket_can_receiver_node.cpp

```diff
@@ -103,22 +102,30 @@ void SocketCanReceiverNode::receive()
   CanId receive_id{};
   can_msgs::msg::Frame frame_msg(rosidl_runtime_cpp::MessageInitialization::ZERO);
   frame_msg.header.frame_id = "can";
-  try {
-    receive_id = receiver_->receive(frame_msg.data.data(), interval_ns_);
-  } catch (const std::exception & ex) {
-    RCLCPP_WARN_THROTTLE(
-      this->get_logger(), *this->get_clock(), 1000,
-      "Error receiving CAN message: %s - %s",
-      interface_.c_str(), ex.what());
-    return;
+
+  while (rclcpp::ok()) {
+    if (this->get_current_state().id() != State::PRIMARY_STATE_ACTIVE) {
+      std::this_thread::sleep_for(500ns);
+      continue;
```

Comment:
I would bet that if you increase this to something like 100ms, the CPU utilization would be essentially 0 and it would be fine because the thread is waiting on the externally-initiated transition.

Reply:
Oh, exactly. Thank you, fixed!
![image](https://user-images.githubusercontent.com/31987104/112258535-00c26180-8caa-11eb-849a-d865c4c016d6.png)